ci/README.md

```diff
@@ -0,0 +1,23 @@
+## CI configurations and examples
+This folder holds vendors CI configurations and examples. Configurations are used to control the vendors CI behaviors, and examples are used as a reference for other vendors to be able to setup a CI of their own.
+
+### Admin list
+The admin list contains the list of github users and organizations that have permission to trigger the vendors CI. Only trusted users who have merge permissions should be on the list. The vendors should be responsible for how the admin list on their CI is updated, but to keep everything organized the vendors CI should at least update their admin list in response to a PR comment with the following phrase `/update-admins`.
+
+### CI Examples
+The examples folder contains configuration examples for vendor CI. It can be used as a reference for vendors to setup their CI. For more information on an example refer to that folder README.
+
+### Vendors CI triggers convention
+A vendor CI trigger phrases should follow the following convention:
+
+```
+/test-<test-type>-<vendor>-<sub-test>
+/skip-<test-type>-<vendor>-<sub-test>
+```
+
+ * `test-type`: The type of test to conduct on the vendor's setups, for example: E2E.
+ * `vendor`: The vendor that implemented the CI.
+ * `sub-test`: In case there are many tests for the `test-type`, this field specify what sub-test to run.
+
+In addition to the convention, all vendors CI should be triggered on the general phrase `/test-all`
+
```

Comment:
there are a couple of things that are unclear to me.

can you elaborate on the expected CI behaviour when <sub-test> is not provided in test/skip and when both <vendor> and <sub-test> is not provided in test/skip ?

also what is the expected behaviour with skip in general ?

Reply:
Nothing would happen, essentially a job gets triggered on a specific trigger phrase, if you replace one section with `all` all the tests for that section will get triggered. 
To elaborate, consider a test called `e2e-nvidia-cx5`, if you want to test it, you should comment one of the following:
 *  `/test-e2e-nvidia-cx5`
 * `/test-e2e-nvidia-all`
 * `/test-e2e-all`
 * `/test-all`

For the skip, it would be a separate job which gets triggered on the `/skip` phrases and the same trigger convention as the `/test` convention above. The job would just `exit 0` to have a green report on the PR with the same `status-context` as the `test` job.

Reply:
> 

Not quite, the patch should be analyze, and determined whether it should be tested or not against the vendor CI. If it should then the `/test` is used to trigger the job and merge will be blocked until the CI is green (either by retesting the test, or uploading a new patchset), ideally the CI would only fail because of a change in the patch and logs should be provided to be able to debug why the patch failed. 
Only when it is determined that there is no need to test this test against this patch, then the `/skip` should be used to pass the test.



> 

well it is not just `device`, it can also be other stuff, say `openshift`, `kind`, and `baremetal` can also be added as a `sub-test`. A note here is that another github workflow will be added after this patch is merged to comment available CI triggers on PR creation.

Reply:
I see, thanks for the explanation!
I think we don't define openshift, kind or baremetal yet, right? they are just possible examples, could be well be a different type (platform for example) in the test- convention.

Reply:
> I think we don't define openshift, kind or baremetal yet, right?

right

> could be well be a different type (platform for example) in the test- convention

yes